Unstable ordering of RPC binding unnamed structures

Ref. https://github.com/nspcc-dev/neo-go/pull/3215, the order is still not solid.
```
--- FAIL: TestAssistedRPCBindings (0.42s)
    --- FAIL: TestAssistedRPCBindings/testdata/rpcbindings/types,_predefined_hash:_false (0.08s)
        generate_test.go:441: 
            	Error Trace:	/home/anna/Documents/GitProjects/nspcc-dev/neo-go/cli/smartcontract/generate_test.go:441
            	Error:      	Not equal: 
            	            	expected: "// Package types contains RPC wrappers for Types contract.\n//\n// Code generated by neo-go contract generate-rpcwrapper --manifest <file.json> --out <file.go> [--hash <hash>] [--config <config>]; DO NOT EDIT.\npackage types\n\nimport (\n\t\"errors\"\n\t\"fmt\"\n\t\"github.com/nspcc-dev/neo-go/pkg/crypto/keys\"\n\t\"github.com/nspcc-dev/neo-go/pkg/neorpc/result\"\n\t\"github.com/nspcc-dev/neo-go/pkg/rpcclient/unwrap\"\n\t\"github.com/nspcc-dev/neo-go/pkg/util\"\n\t\"github.com/nspcc-dev/neo-go/pkg/vm/stackitem\"\n\t\"math/big\"\n\t\"unicode/utf8\"\n)\n\n// Unnamed is a contract-specific unnamed type used by its methods.\ntype Unnamed struct {\n\tI *big.Int\n}\n\n// UnnamedX is a contract-specific unnamedX type used by its methods.\ntype UnnamedX struct {\n\tI *big.Int\n\tB bool\n}\n\n// Invoker is used by ContractReader to call various safe methods.\ntype Invoker interface {\n\tCall(contract util.Uint160, operation string, params ...any) (*result.Invoke, error)\n}\n\n// ContractReader implements safe contract methods.\ntype ContractReader struct {\n\tinvoker Invoker\n\thash    util.Uint160\n}\n\n// NewReader creates an instance of ContractReader using provided contract hash and the given Invoker.\nfunc NewReader(invoker Invoker, hash util.Uint160) *ContractReader {\n\treturn &ContractReader{invoker, hash}\n}\n\n// AAAStrings invokes `aAAStrings` method of contract.\nfunc (c *ContractReader) AAAStrings(s [][][]string) ([][][]string, error) {\n\treturn func(item stackitem.Item, err error) ([][][]string, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) ([][][]string, error) {\n\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\tif !ok {\n\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t}\n\t\t\tres := make([][][]string, len(arr))\n\t\t\tfor i := range res {\n\t\t\t\tres[i], err = func(item stackitem.Item) ([][]string, error) {\n\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\tif !ok {\n\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t}\n\t\t\t\t\tres := make([][]string, len(arr))\n\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\tres[i], err = func(item stackitem.Item) ([]string, error) {\n\t\t\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\t\t\tif !ok {\n\t\t\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\tres := make([]string, len(arr))\n\t\t\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\t\t\tres[i], err = func(item stackitem.Item) (string, error) {\n\t\t\t\t\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\treturn string(b), nil\n\t\t\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\treturn res, nil\n\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t\treturn res, nil\n\t\t\t\t}(arr[i])\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"aAAStrings\", s)))\n}\n\n// Any invokes `any` method of contract.\nfunc (c *ContractReader) Any(a any) (any, error) {\n\treturn func(item stackitem.Item, err error) (any, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn item.Value(), error(nil)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"any\", a)))\n}\n\n// AnyMaps invokes `anyMaps` method of contract.\nfunc (c *ContractReader) AnyMaps(m map[*big.Int]any) (map[*big.Int]any, error) {\n\treturn func(item stackitem.Item, err error) (map[*big.Int]any, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) (map[*big.Int]any, error) {\n\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\tif !ok {\n\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t}\n\t\t\tres := make(map[*big.Int]any)\n\t\t\tfor i := range m {\n\t\t\t\tk, err := m[i].Key.TryInteger()\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tv, err := m[i].Value.Value(), error(nil)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tres[k] = v\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"anyMaps\", m)))\n}\n\n// Bool invokes `bool` method of contract.\nfunc (c *ContractReader) Bool(b bool) (bool, error) {\n\treturn unwrap.Bool(c.invoker.Call(c.hash, \"bool\", b))\n}\n\n// Bools invokes `bools` method of contract.\nfunc (c *ContractReader) Bools(b []bool) ([]bool, error) {\n\treturn unwrap.ArrayOfBools(c.invoker.Call(c.hash, \"bools\", b))\n}\n\n// Bytes invokes `bytes` method of contract.\nfunc (c *ContractReader) Bytes(b []byte) ([]byte, error) {\n\treturn unwrap.Bytes(c.invoker.Call(c.hash, \"bytes\", b))\n}\n\n// Bytess invokes `bytess` method of contract.\nfunc (c *ContractReader) Bytess(b [][]byte) ([][]byte, error) {\n\treturn unwrap.ArrayOfBytes(c.invoker.Call(c.hash, \"bytess\", b))\n}\n\n// CrazyMaps invokes `crazyMaps` method of contract.\nfunc (c *ContractReader) CrazyMaps(m map[*big.Int][]map[string][]util.Uint160) (map[*big.Int][]map[string][]util.Uint160, error) {\n\treturn func(item stackitem.Item, err error) (map[*big.Int][]map[string][]util.Uint160, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) (map[*big.Int][]map[string][]util.Uint160, error) {\n\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\tif !ok {\n\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t}\n\t\t\tres := make(map[*big.Int][]map[string][]util.Uint160)\n\t\t\tfor i := range m {\n\t\t\t\tk, err := m[i].Key.TryInteger()\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tv, err := func(item stackitem.Item) ([]map[string][]util.Uint160, error) {\n\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\tif !ok {\n\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t}\n\t\t\t\t\tres := make([]map[string][]util.Uint160, len(arr))\n\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\tres[i], err = func(item stackitem.Item) (map[string][]util.Uint160, error) {\n\t\t\t\t\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\t\t\t\t\tif !ok {\n\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\tres := make(map[string][]util.Uint160)\n\t\t\t\t\t\t\tfor i := range m {\n\t\t\t\t\t\t\t\tk, err := func(item stackitem.Item) (string, error) {\n\t\t\t\t\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\treturn string(b), nil\n\t\t\t\t\t\t\t\t}(m[i].Key)\n\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\tv, err := func(item stackitem.Item) ([]util.Uint160, error) {\n\t\t\t\t\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\t\t\t\t\tif !ok {\n\t\t\t\t\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\tres := make([]util.Uint160, len(arr))\n\t\t\t\t\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\t\t\t\t\tres[i], err = func(item stackitem.Item) (util.Uint160, error) {\n\t\t\t\t\t\t\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\t\t\treturn util.Uint160{}, err\n\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\tu, err := util.Uint160DecodeBytesBE(b)\n\t\t\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\t\t\treturn util.Uint160{}, err\n\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\treturn u, nil\n\t\t\t\t\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\treturn res, nil\n\t\t\t\t\t\t\t\t}(m[i].Value)\n\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\tres[k] = v\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\treturn res, nil\n\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t\treturn res, nil\n\t\t\t\t}(m[i].Value)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tres[k] = v\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"crazyMaps\", m)))\n}\n\n// Hash160 invokes `hash160` method of contract.\nfunc (c *ContractReader) Hash160(h util.Uint160) (util.Uint160, error) {\n\treturn unwrap.Uint160(c.invoker.Call(c.hash, \"hash160\", h))\n}\n\n// Hash160s invokes `hash160s` method of contract.\nfunc (c *ContractReader) Hash160s(h []util.Uint160) ([]util.Uint160, error) {\n\treturn unwrap.ArrayOfUint160(c.invoker.Call(c.hash, \"hash160s\", h))\n}\n\n// Hash256 invokes `hash256` method of contract.\nfunc (c *ContractReader) Hash256(h util.Uint256) (util.Uint256, error) {\n\treturn unwrap.Uint256(c.invoker.Call(c.hash, \"hash256\", h))\n}\n\n// Hash256s invokes `hash256s` method of contract.\nfunc (c *ContractReader) Hash256s(h []util.Uint256) ([]util.Uint256, error) {\n\treturn unwrap.ArrayOfUint256(c.invoker.Call(c.hash, \"hash256s\", h))\n}\n\n// Int invokes `int` method of contract.\nfunc (c *ContractReader) Int(i *big.Int) (*big.Int, error) {\n\treturn unwrap.BigInt(c.invoker.Call(c.hash, \"int\", i))\n}\n\n// Ints invokes `ints` method of contract.\nfunc (c *ContractReader) Ints(i []*big.Int) ([]*big.Int, error) {\n\treturn unwrap.ArrayOfBigInts(c.invoker.Call(c.hash, \"ints\", i))\n}\n\n// Maps invokes `maps` method of contract.\nfunc (c *ContractReader) Maps(m map[string]string) (map[string]string, error) {\n\treturn func(item stackitem.Item, err error) (map[string]string, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) (map[string]string, error) {\n\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\tif !ok {\n\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t}\n\t\t\tres := make(map[string]string)\n\t\t\tfor i := range m {\n\t\t\t\tk, err := func(item stackitem.Item) (string, error) {\n\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\tif err != nil {\n\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t}\n\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t}\n\t\t\t\t\treturn string(b), nil\n\t\t\t\t}(m[i].Key)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tv, err := func(item stackitem.Item) (string, error) {\n\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\tif err != nil {\n\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t}\n\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t}\n\t\t\t\t\treturn string(b), nil\n\t\t\t\t}(m[i].Value)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tres[k] = v\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"maps\", m)))\n}\n\n// PublicKey invokes `publicKey` method of contract.\nfunc (c *ContractReader) PublicKey(k *keys.PublicKey) (*keys.PublicKey, error) {\n\treturn unwrap.PublicKey(c.invoker.Call(c.hash, \"publicKey\", k))\n}\n\n// PublicKeys invokes `publicKeys` method of contract.\nfunc (c *ContractReader) PublicKeys(k keys.PublicKeys) (keys.PublicKeys, error) {\n\treturn unwrap.ArrayOfPublicKeys(c.invoker.Call(c.hash, \"publicKeys\", k))\n}\n\n// Signature invokes `signature` method of contract.\nfunc (c *ContractReader) Signature(s []byte) ([]byte, error) {\n\treturn unwrap.Bytes(c.invoker.Call(c.hash, \"signature\", s))\n}\n\n// Signatures invokes `signatures` method of contract.\nfunc (c *ContractReader) Signatures(s [][]byte) ([][]byte, error) {\n\treturn unwrap.ArrayOfBytes(c.invoker.Call(c.hash, \"signatures\", s))\n}\n\n// String invokes `string` method of contract.\nfunc (c *ContractReader) String(s string) (string, error) {\n\treturn unwrap.UTF8String(c.invoker.Call(c.hash, \"string\", s))\n}\n\n// Strings invokes `strings` method of contract.\nfunc (c *ContractReader) Strings(s []string) ([]string, error) {\n\treturn unwrap.ArrayOfUTF8Strings(c.invoker.Call(c.hash, \"strings\", s))\n}\n\n// UnnamedStructs invokes `unnamedStructs` method of contract.\nfunc (c *ContractReader) UnnamedStructs() (*Unnamed, error) {\n\treturn itemToUnnamed(unwrap.Item(c.invoker.Call(c.hash, \"unnamedStructs\")))\n}\n\n// UnnamedStructsX invokes `unnamedStructsX` method of contract.\nfunc (c *ContractReader) UnnamedStructsX() (*UnnamedX, error) {\n\treturn itemToUnnamedX(unwrap.Item(c.invoker.Call(c.hash, \"unnamedStructsX\")))\n}\n\n// itemToUnnamed converts stack item into *Unnamed.\nfunc itemToUnnamed(item stackitem.Item, err error) (*Unnamed, error) {\n\tif err != nil {\n\t\treturn nil, err\n\t}\n\tvar res = new(Unnamed)\n\terr = res.FromStackItem(item)\n\treturn res, err\n}\n\n// FromStackItem retrieves fields of Unnamed from the given\n// [stackitem.Item] or returns an error if it's not possible to do to so.\nfunc (res *Unnamed) FromStackItem(item stackitem.Item) error {\n\tarr, ok := item.Value().([]stackitem.Item)\n\tif !ok {\n\t\treturn errors.New(\"not an array\")\n\t}\n\tif len(arr) != 1 {\n\t\treturn errors.New(\"wrong number of structure elements\")\n\t}\n\n\tvar (\n\t\tindex = -1\n\t\terr   error\n\t)\n\tindex++\n\tres.I, err = arr[index].TryInteger()\n\tif err != nil {\n\t\treturn fmt.Errorf(\"field I: %w\", err)\n\t}\n\n\treturn nil\n}\n\n// itemToUnnamedX converts stack item into *UnnamedX.\nfunc itemToUnnamedX(item stackitem.Item, err error) (*UnnamedX, error) {\n\tif err != nil {\n\t\treturn nil, err\n\t}\n\tvar res = new(UnnamedX)\n\terr = res.FromStackItem(item)\n\treturn res, err\n}\n\n// FromStackItem retrieves fields of UnnamedX from the given\n// [stackitem.Item] or returns an error if it's not possible to do to so.\nfunc (res *UnnamedX) FromStackItem(item stackitem.Item) error {\n\tarr, ok := item.Value().([]stackitem.Item)\n\tif !ok {\n\t\treturn errors.New(\"not an array\")\n\t}\n\tif len(arr) != 2 {\n\t\treturn errors.New(\"wrong number of structure elements\")\n\t}\n\n\tvar (\n\t\tindex = -1\n\t\terr   error\n\t)\n\tindex++\n\tres.I, err = arr[index].TryInteger()\n\tif err != nil {\n\t\treturn fmt.Errorf(\"field I: %w\", err)\n\t}\n\n\tindex++\n\tres.B, err = arr[index].TryBool()\n\tif err != nil {\n\t\treturn fmt.Errorf(\"field B: %w\", err)\n\t}\n\n\treturn nil\n}\n"
            	            	actual  : "// Package types contains RPC wrappers for Types contract.\n//\n// Code generated by neo-go contract generate-rpcwrapper --manifest <file.json> --out <file.go> [--hash <hash>] [--config <config>]; DO NOT EDIT.\npackage types\n\nimport (\n\t\"errors\"\n\t\"fmt\"\n\t\"github.com/nspcc-dev/neo-go/pkg/crypto/keys\"\n\t\"github.com/nspcc-dev/neo-go/pkg/neorpc/result\"\n\t\"github.com/nspcc-dev/neo-go/pkg/rpcclient/unwrap\"\n\t\"github.com/nspcc-dev/neo-go/pkg/util\"\n\t\"github.com/nspcc-dev/neo-go/pkg/vm/stackitem\"\n\t\"math/big\"\n\t\"unicode/utf8\"\n)\n\n// Unnamed is a contract-specific unnamed type used by its methods.\ntype Unnamed struct {\n\tI *big.Int\n\tB bool\n}\n\n// UnnamedX is a contract-specific unnamedX type used by its methods.\ntype UnnamedX struct {\n\tI *big.Int\n}\n\n// Invoker is used by ContractReader to call various safe methods.\ntype Invoker interface {\n\tCall(contract util.Uint160, operation string, params ...any) (*result.Invoke, error)\n}\n\n// ContractReader implements safe contract methods.\ntype ContractReader struct {\n\tinvoker Invoker\n\thash    util.Uint160\n}\n\n// NewReader creates an instance of ContractReader using provided contract hash and the given Invoker.\nfunc NewReader(invoker Invoker, hash util.Uint160) *ContractReader {\n\treturn &ContractReader{invoker, hash}\n}\n\n// AAAStrings invokes `aAAStrings` method of contract.\nfunc (c *ContractReader) AAAStrings(s [][][]string) ([][][]string, error) {\n\treturn func(item stackitem.Item, err error) ([][][]string, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) ([][][]string, error) {\n\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\tif !ok {\n\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t}\n\t\t\tres := make([][][]string, len(arr))\n\t\t\tfor i := range res {\n\t\t\t\tres[i], err = func(item stackitem.Item) ([][]string, error) {\n\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\tif !ok {\n\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t}\n\t\t\t\t\tres := make([][]string, len(arr))\n\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\tres[i], err = func(item stackitem.Item) ([]string, error) {\n\t\t\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\t\t\tif !ok {\n\t\t\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\tres := make([]string, len(arr))\n\t\t\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\t\t\tres[i], err = func(item stackitem.Item) (string, error) {\n\t\t\t\t\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\treturn string(b), nil\n\t\t\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\treturn res, nil\n\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t\treturn res, nil\n\t\t\t\t}(arr[i])\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"aAAStrings\", s)))\n}\n\n// Any invokes `any` method of contract.\nfunc (c *ContractReader) Any(a any) (any, error) {\n\treturn func(item stackitem.Item, err error) (any, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn item.Value(), error(nil)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"any\", a)))\n}\n\n// AnyMaps invokes `anyMaps` method of contract.\nfunc (c *ContractReader) AnyMaps(m map[*big.Int]any) (map[*big.Int]any, error) {\n\treturn func(item stackitem.Item, err error) (map[*big.Int]any, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) (map[*big.Int]any, error) {\n\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\tif !ok {\n\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t}\n\t\t\tres := make(map[*big.Int]any)\n\t\t\tfor i := range m {\n\t\t\t\tk, err := m[i].Key.TryInteger()\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tv, err := m[i].Value.Value(), error(nil)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tres[k] = v\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"anyMaps\", m)))\n}\n\n// Bool invokes `bool` method of contract.\nfunc (c *ContractReader) Bool(b bool) (bool, error) {\n\treturn unwrap.Bool(c.invoker.Call(c.hash, \"bool\", b))\n}\n\n// Bools invokes `bools` method of contract.\nfunc (c *ContractReader) Bools(b []bool) ([]bool, error) {\n\treturn unwrap.ArrayOfBools(c.invoker.Call(c.hash, \"bools\", b))\n}\n\n// Bytes invokes `bytes` method of contract.\nfunc (c *ContractReader) Bytes(b []byte) ([]byte, error) {\n\treturn unwrap.Bytes(c.invoker.Call(c.hash, \"bytes\", b))\n}\n\n// Bytess invokes `bytess` method of contract.\nfunc (c *ContractReader) Bytess(b [][]byte) ([][]byte, error) {\n\treturn unwrap.ArrayOfBytes(c.invoker.Call(c.hash, \"bytess\", b))\n}\n\n// CrazyMaps invokes `crazyMaps` method of contract.\nfunc (c *ContractReader) CrazyMaps(m map[*big.Int][]map[string][]util.Uint160) (map[*big.Int][]map[string][]util.Uint160, error) {\n\treturn func(item stackitem.Item, err error) (map[*big.Int][]map[string][]util.Uint160, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) (map[*big.Int][]map[string][]util.Uint160, error) {\n\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\tif !ok {\n\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t}\n\t\t\tres := make(map[*big.Int][]map[string][]util.Uint160)\n\t\t\tfor i := range m {\n\t\t\t\tk, err := m[i].Key.TryInteger()\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tv, err := func(item stackitem.Item) ([]map[string][]util.Uint160, error) {\n\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\tif !ok {\n\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t}\n\t\t\t\t\tres := make([]map[string][]util.Uint160, len(arr))\n\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\tres[i], err = func(item stackitem.Item) (map[string][]util.Uint160, error) {\n\t\t\t\t\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\t\t\t\t\tif !ok {\n\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\tres := make(map[string][]util.Uint160)\n\t\t\t\t\t\t\tfor i := range m {\n\t\t\t\t\t\t\t\tk, err := func(item stackitem.Item) (string, error) {\n\t\t\t\t\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\treturn string(b), nil\n\t\t\t\t\t\t\t\t}(m[i].Key)\n\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\tv, err := func(item stackitem.Item) ([]util.Uint160, error) {\n\t\t\t\t\t\t\t\t\tarr, ok := item.Value().([]stackitem.Item)\n\t\t\t\t\t\t\t\t\tif !ok {\n\t\t\t\t\t\t\t\t\t\treturn nil, errors.New(\"not an array\")\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\tres := make([]util.Uint160, len(arr))\n\t\t\t\t\t\t\t\t\tfor i := range res {\n\t\t\t\t\t\t\t\t\t\tres[i], err = func(item stackitem.Item) (util.Uint160, error) {\n\t\t\t\t\t\t\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\t\t\treturn util.Uint160{}, err\n\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\tu, err := util.Uint160DecodeBytesBE(b)\n\t\t\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\t\t\treturn util.Uint160{}, err\n\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\treturn u, nil\n\t\t\t\t\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\treturn res, nil\n\t\t\t\t\t\t\t\t}(m[i].Value)\n\t\t\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\tres[k] = v\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\treturn res, nil\n\t\t\t\t\t\t}(arr[i])\n\t\t\t\t\t\tif err != nil {\n\t\t\t\t\t\t\treturn nil, fmt.Errorf(\"item %d: %w\", i, err)\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t\treturn res, nil\n\t\t\t\t}(m[i].Value)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tres[k] = v\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"crazyMaps\", m)))\n}\n\n// Hash160 invokes `hash160` method of contract.\nfunc (c *ContractReader) Hash160(h util.Uint160) (util.Uint160, error) {\n\treturn unwrap.Uint160(c.invoker.Call(c.hash, \"hash160\", h))\n}\n\n// Hash160s invokes `hash160s` method of contract.\nfunc (c *ContractReader) Hash160s(h []util.Uint160) ([]util.Uint160, error) {\n\treturn unwrap.ArrayOfUint160(c.invoker.Call(c.hash, \"hash160s\", h))\n}\n\n// Hash256 invokes `hash256` method of contract.\nfunc (c *ContractReader) Hash256(h util.Uint256) (util.Uint256, error) {\n\treturn unwrap.Uint256(c.invoker.Call(c.hash, \"hash256\", h))\n}\n\n// Hash256s invokes `hash256s` method of contract.\nfunc (c *ContractReader) Hash256s(h []util.Uint256) ([]util.Uint256, error) {\n\treturn unwrap.ArrayOfUint256(c.invoker.Call(c.hash, \"hash256s\", h))\n}\n\n// Int invokes `int` method of contract.\nfunc (c *ContractReader) Int(i *big.Int) (*big.Int, error) {\n\treturn unwrap.BigInt(c.invoker.Call(c.hash, \"int\", i))\n}\n\n// Ints invokes `ints` method of contract.\nfunc (c *ContractReader) Ints(i []*big.Int) ([]*big.Int, error) {\n\treturn unwrap.ArrayOfBigInts(c.invoker.Call(c.hash, \"ints\", i))\n}\n\n// Maps invokes `maps` method of contract.\nfunc (c *ContractReader) Maps(m map[string]string) (map[string]string, error) {\n\treturn func(item stackitem.Item, err error) (map[string]string, error) {\n\t\tif err != nil {\n\t\t\treturn nil, err\n\t\t}\n\t\treturn func(item stackitem.Item) (map[string]string, error) {\n\t\t\tm, ok := item.Value().([]stackitem.MapElement)\n\t\t\tif !ok {\n\t\t\t\treturn nil, fmt.Errorf(\"%s is not a map\", item.Type().String())\n\t\t\t}\n\t\t\tres := make(map[string]string)\n\t\t\tfor i := range m {\n\t\t\t\tk, err := func(item stackitem.Item) (string, error) {\n\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\tif err != nil {\n\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t}\n\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t}\n\t\t\t\t\treturn string(b), nil\n\t\t\t\t}(m[i].Key)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"key %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tv, err := func(item stackitem.Item) (string, error) {\n\t\t\t\t\tb, err := item.TryBytes()\n\t\t\t\t\tif err != nil {\n\t\t\t\t\t\treturn \"\", err\n\t\t\t\t\t}\n\t\t\t\t\tif !utf8.Valid(b) {\n\t\t\t\t\t\treturn \"\", errors.New(\"not a UTF-8 string\")\n\t\t\t\t\t}\n\t\t\t\t\treturn string(b), nil\n\t\t\t\t}(m[i].Value)\n\t\t\t\tif err != nil {\n\t\t\t\t\treturn nil, fmt.Errorf(\"value %d: %w\", i, err)\n\t\t\t\t}\n\t\t\t\tres[k] = v\n\t\t\t}\n\t\t\treturn res, nil\n\t\t}(item)\n\t}(unwrap.Item(c.invoker.Call(c.hash, \"maps\", m)))\n}\n\n// PublicKey invokes `publicKey` method of contract.\nfunc (c *ContractReader) PublicKey(k *keys.PublicKey) (*keys.PublicKey, error) {\n\treturn unwrap.PublicKey(c.invoker.Call(c.hash, \"publicKey\", k))\n}\n\n// PublicKeys invokes `publicKeys` method of contract.\nfunc (c *ContractReader) PublicKeys(k keys.PublicKeys) (keys.PublicKeys, error) {\n\treturn unwrap.ArrayOfPublicKeys(c.invoker.Call(c.hash, \"publicKeys\", k))\n}\n\n// Signature invokes `signature` method of contract.\nfunc (c *ContractReader) Signature(s []byte) ([]byte, error) {\n\treturn unwrap.Bytes(c.invoker.Call(c.hash, \"signature\", s))\n}\n\n// Signatures invokes `signatures` method of contract.\nfunc (c *ContractReader) Signatures(s [][]byte) ([][]byte, error) {\n\treturn unwrap.ArrayOfBytes(c.invoker.Call(c.hash, \"signatures\", s))\n}\n\n// String invokes `string` method of contract.\nfunc (c *ContractReader) String(s string) (string, error) {\n\treturn unwrap.UTF8String(c.invoker.Call(c.hash, \"string\", s))\n}\n\n// Strings invokes `strings` method of contract.\nfunc (c *ContractReader) Strings(s []string) ([]string, error) {\n\treturn unwrap.ArrayOfUTF8Strings(c.invoker.Call(c.hash, \"strings\", s))\n}\n\n// UnnamedStructs invokes `unnamedStructs` method of contract.\nfunc (c *ContractReader) UnnamedStructs() (*UnnamedX, error) {\n\treturn itemToUnnamedX(unwrap.Item(c.invoker.Call(c.hash, \"unnamedStructs\")))\n}\n\n// UnnamedStructsX invokes `unnamedStructsX` method of contract.\nfunc (c *ContractReader) UnnamedStructsX() (*Unnamed, error) {\n\treturn itemToUnnamed(unwrap.Item(c.invoker.Call(c.hash, \"unnamedStructsX\")))\n}\n\n// itemToUnnamed converts stack item into *Unnamed.\nfunc itemToUnnamed(item stackitem.Item, err error) (*Unnamed, error) {\n\tif err != nil {\n\t\treturn nil, err\n\t}\n\tvar res = new(Unnamed)\n\terr = res.FromStackItem(item)\n\treturn res, err\n}\n\n// FromStackItem retrieves fields of Unnamed from the given\n// [stackitem.Item] or returns an error if it's not possible to do to so.\nfunc (res *Unnamed) FromStackItem(item stackitem.Item) error {\n\tarr, ok := item.Value().([]stackitem.Item)\n\tif !ok {\n\t\treturn errors.New(\"not an array\")\n\t}\n\tif len(arr) != 2 {\n\t\treturn errors.New(\"wrong number of structure elements\")\n\t}\n\n\tvar (\n\t\tindex = -1\n\t\terr   error\n\t)\n\tindex++\n\tres.I, err = arr[index].TryInteger()\n\tif err != nil {\n\t\treturn fmt.Errorf(\"field I: %w\", err)\n\t}\n\n\tindex++\n\tres.B, err = arr[index].TryBool()\n\tif err != nil {\n\t\treturn fmt.Errorf(\"field B: %w\", err)\n\t}\n\n\treturn nil\n}\n\n// itemToUnnamedX converts stack item into *UnnamedX.\nfunc itemToUnnamedX(item stackitem.Item, err error) (*UnnamedX, error) {\n\tif err != nil {\n\t\treturn nil, err\n\t}\n\tvar res = new(UnnamedX)\n\terr = res.FromStackItem(item)\n\treturn res, err\n}\n\n// FromStackItem retrieves fields of UnnamedX from the given\n// [stackitem.Item] or returns an error if it's not possible to do to so.\nfunc (res *UnnamedX) FromStackItem(item stackitem.Item) error {\n\tarr, ok := item.Value().([]stackitem.Item)\n\tif !ok {\n\t\treturn errors.New(\"not an array\")\n\t}\n\tif len(arr) != 1 {\n\t\treturn errors.New(\"wrong number of structure elements\")\n\t}\n\n\tvar (\n\t\tindex = -1\n\t\terr   error\n\t)\n\tindex++\n\tres.I, err = arr[index].TryInteger()\n\tif err != nil {\n\t\treturn fmt.Errorf(\"field I: %w\", err)\n\t}\n\n\treturn nil\n}\n"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -20,2 +20,3 @@
            	            	 	I *big.Int
            	            	+	B bool
            	            	 }
            	            	@@ -25,3 +26,2 @@
            	            	 	I *big.Int
            	            	-	B bool
            	            	 }
            	            	@@ -358,4 +358,4 @@
            	            	 // UnnamedStructs invokes `unnamedStructs` method of contract.
            	            	-func (c *ContractReader) UnnamedStructs() (*Unnamed, error) {
            	            	-	return itemToUnnamed(unwrap.Item(c.invoker.Call(c.hash, "unnamedStructs")))
            	            	+func (c *ContractReader) UnnamedStructs() (*UnnamedX, error) {
            	            	+	return itemToUnnamedX(unwrap.Item(c.invoker.Call(c.hash, "unnamedStructs")))
            	            	 }
            	            	@@ -363,4 +363,4 @@
            	            	 // UnnamedStructsX invokes `unnamedStructsX` method of contract.
            	            	-func (c *ContractReader) UnnamedStructsX() (*UnnamedX, error) {
            	            	-	return itemToUnnamedX(unwrap.Item(c.invoker.Call(c.hash, "unnamedStructsX")))
            	            	+func (c *ContractReader) UnnamedStructsX() (*Unnamed, error) {
            	            	+	return itemToUnnamed(unwrap.Item(c.invoker.Call(c.hash, "unnamedStructsX")))
            	            	 }
            	            	@@ -384,3 +384,3 @@
            	            	 	}
            	            	-	if len(arr) != 1 {
            	            	+	if len(arr) != 2 {
            	            	 		return errors.New("wrong number of structure elements")
            	            	@@ -398,2 +398,8 @@
            	            	 
            	            	+	index++
            	            	+	res.B, err = arr[index].TryBool()
            	            	+	if err != nil {
            	            	+		return fmt.Errorf("field B: %w", err)
            	            	+	}
            	            	+
            	            	 	return nil
            	            	@@ -418,3 +424,3 @@
            	            	 	}
            	            	-	if len(arr) != 2 {
            	            	+	if len(arr) != 1 {
            	            	 		return errors.New("wrong number of structure elements")
            	            	@@ -432,8 +438,2 @@
            	            	 
            	            	-	index++
            	            	-	res.B, err = arr[index].TryBool()
            	            	-	if err != nil {
            	            	-		return fmt.Errorf("field B: %w", err)
            	            	-	}
            	            	-
            	            	 	return nil
            	Test:       	TestAssistedRPCBindings/testdata/rpcbindings/types,_predefined_hash:_false
```

--- a/pkg/compiler/debug.go
+++ b/pkg/compiler/debug.go
@@ -203,18 +203,19 @@ func (c *codegen) emitDebugInfo(contract []byte) *DebugInfo {
 		})
 	}
 
-	start := len(d.Methods)
-	d.NamedTypes = make(map[string]binding.ExtendedType)
+	var fnames = make([]string, 0, len(c.funcs))
 	for name, scope := range c.funcs {
 		if scope.rng.Start == scope.rng.End {
 			continue
 		}
-		m := c.methodInfoFromScope(name, scope, d.NamedTypes)
+		fnames = append(fnames, name)
+	}
+	sort.Strings(fnames)
+	d.NamedTypes = make(map[string]binding.ExtendedType)
+	for _, name := range fnames {
+		m := c.methodInfoFromScope(name, c.funcs[name], d.NamedTypes)
 		d.Methods = append(d.Methods, *m)
 	}
-	sort.Slice(d.Methods[start:], func(i, j int) bool {
-		return d.Methods[start+i].Name.Name < d.Methods[start+j].Name.Name
-	})
 	d.EmittedEvents = c.emittedEvents
 	d.InvokedContracts = c.invokedContracts
 	return d

